docs/pageserver-thread-mgmt.md

```diff
@@ -21,19 +26,82 @@ also a `shudown_watcher()` Future that can be used with `tokio::select!`
 or similar, to wake up on shutdown.
 
 
-### Sync vs async
+### Async cancellation safety
+
+In async Rust, futures can be "cancelled" at any await point, by
+dropping the Future. For example, `tokio::select!` returns as soon as
+one of the Futures returns, and drops the others. In the Rust
+ecosystem, some functions are cancellation-safe, meaning they can be
+safely dropped without side-effects, while others are not. See
+documentation of `tokio::select!` for examples.
+
+In the pageserver and safekeeper, async code is *not*
+cancellation-safe by default. You can assume that any async code you
+write is polled to completion. There are a few exceptions to this,
+like the code that handles incoming HTTP requests, because Hyper will
+drop the request-handling Future if the HTTP connection is lost. The
```

Comment:
We haven''t actually done that "spawn a new task for each request" yet, but we should

Reply:
Let's fix that before we merge this PR, or, create an issue and link to it from here.